### PR TITLE
Firebase authentication

### PR DESCRIPTION
**Pull request by MuskanChhatrasal**
Added firebase authentication

### CODE REVIEW

File: src/Components/Navbar/navbar.js [hunk 1]
```diff
@@ -1,7 +1,36 @@
...
+import { signOut, onAuthStateChanged } from "firebase/auth";
+import { auth } from "../../firebase";
 
 const Navbar = () => {
+  const [user, setUser] = useState({});
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (currentUser) => {
+      console.log(currentUser);
```
Comment (BalajiHarinath) [inline src/Components/Navbar/navbar.js:12]: Can remove the console log if not required

File: src/Components/Navbar/navbar.js [hunk 2]
```diff
@@ -15,16 +44,50 @@ const Navbar = () => {
...
+            >
+              <i
+                class="fas fa-sign-out-alt"
+                style={{
+                  marginRight: "0.5rem",
+                  color: "black",
+                }}
+              ></i>
+              Logout
             </a>
```
Comment (BalajiHarinath) [inline src/Components/Navbar/navbar.js:74]: Suggestion: Can use a button instead of anchor tag

File: src/Components/ProtectedRoute.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React, { useState, useEffect } from "react";
+import { Navigate, Outlet } from "react-router-dom";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "../firebase";
+
+const ProtectedRoute = () => {
+  const [user, setUser] = useState({});
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (currentUser) => {
+      console.log(currentUser);
```
Comment (BalajiHarinath) [inline src/Components/ProtectedRoute.js:10]: Can remove the console log if not required

File: src/Pages/Home/home.js [hunk 4]
```diff
@@ -1,8 +1,29 @@
...
+import { auth } from "../../firebase";
+import { async } from "@firebase/util";
 
 const Home = () => {
+  const [user, setUser] = useState({});
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (currentUser) => {
+      console.log(currentUser);
```
Comment (BalajiHarinath) [inline src/Pages/Home/home.js:19]: Can remove the console log if not required

**Review by BalajiHarinath — APPROVED** [review]
Apart from the suggestions everything looks good.

File: src/Components/Navbar/navbar.js [hunk 5]
```diff
@@ -15,16 +44,50 @@ const Navbar = () => {
...
-        <li style={{ marginTop: "0.25rem", cursor: "pointer" }}>
-          <li style={{ cursor: "pointer" }}>
-            <a href="https://github.com/MuskanChhatrasal">
-              <i style={{ marginRight: "0.5rem" }}></i>
-              Github
+        <li style={{ marginTop: "0.75rem", cursor: "pointer" }}>
+          {!user ? <i class="fas fa-user-alt"></i> : `Hello, ${user.email}`}
+        </li>
+        <li
+          style={{
```
Comment (aayushjain-code) [inline src/Components/Navbar/navbar.js:53]: Try To avoid inline styling

**Review by aayushjain-code — APPROVED** [review]
You Can merge it it looks nice.

File: src/Components/Navbar/navbar.js (hunk 1 above)
Comment (MuskanChhatrasal) [reply src/Components/Navbar/navbar.js:12]: Thankyou for reviewing. Will surely make the suggested changes

**Pull request merged**